File: MarxUp/MarxUp/DataSources/ToolboxItemCollectionDataSource/ArrowsCollectionViewDataSource.m
//
//  ArrowsCollectionViewDataSource.m
//  MarxUp
//
//  Created by Ognyanka Boneva on 1.10.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

#import "ArrowsCollectionViewDataSource.h"
#import "ToolboxItemOptionsCollectionViewCell.h"
#import "ArrowModel.h"
#import "Constants.h"


@interface ArrowsCollectionViewDataSource ()

@property (strong, nonatomic)NSArray<ArrowModel *> *arrows;

@end

@implementation ArrowsCollectionViewDataSource


+ (instancetype)newDataSource {
    ArrowsCollectionViewDataSource *newDataSource = [ArrowsCollectionViewDataSource new];
    
    ArrowModel *openArrow = [ArrowModel newArrowFromType:ArrowEndLineTypeOpen];
    ArrowModel *closedArrow = [ArrowModel newArrowFromType:ArrowEndLineTypeClosed];
    newDataSource.arrows = @[openArrow, closedArrow];
    
    return newDataSource;
}

- (ArrowEndLineType)optionAtIndex:(NSInteger)index {
    return self.arrows[index].type;
}

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return self.arrows.count;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    ToolboxItemOptionsCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:IDENTIFIER_TOOLBOX_ITEM_OPTIONS_CELL forIndexPath:indexPath];
    ArrowModel *model = self.arrows[indexPath.item];
    
    cell.imageView.image = model.image;
    
    return cell;
}

@end

File: MarxUpUp/MarxUpUp/DataSources/ToolboxItemCollectionDataSource/ArrowsCollectionViewDataSource.swift
//
//  ArrowsCollectionViewDataSource.swift
//  MarxUpUp
//
//  Created by Ognyanka Boneva on 14.11.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

import UIKit

final class ArrowsCollectionViewDataSource: NSObject {
    private let arrows = [ArrowModel(withType: ArrowEndLineType.Open), ArrowModel(withType: ArrowEndLineType.Closed)]
}

extension ArrowsCollectionViewDataSource: ToolboxItemCollectionDataSource {
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return arrows.count
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(fromClass: ToolboxItemOptionsCollectionViewCell.self, forIndexPath: indexPath)
        
        cell.itemImageView.image = arrows[indexPath.item].image
        
        return cell
    }
    
    func option(atIndex index: Int) -> Int {
        return arrows[index].type.rawValue
    }
    
    func numberOfSections(in collectionView: UICollectionView) -> Int {
        return 1
    }
}
